icu4c/source/data/BUILDRULES.py

```diff
@@ -517,18 +516,18 @@ def generate_tree(
     ]
 
     # Generate res_index file
-    synthetic_locales = set()
-    if not config.ignore_xml_deprecates:
-        deprecates_xml_path = os.path.join(
-            os.path.dirname(__file__), xml_filename)
-        deprecates_xml = ET.parse(deprecates_xml_path)
-        for child in deprecates_xml.getroot():
-            if child.tag == "alias":
-                synthetic_locales.add(child.attrib["from"])
-            elif child.tag == "emptyLocale":
-                synthetic_locales.add(child.attrib["locale"])
-            else:
-                raise ValueError("Unknown tag in deprecates XML: %s" % child.tag)
+    # Exclude the deprecated locale variants; see ICU-20628
+    synthetic_locales = set([
```

Comment:
should be data driven, better to add something to the xml

Reply:
Part of the mechanical purpose of this change is to *avoid* having to read any XML file in Python.  It's extremely unlikely that this list will grow, and hopefully these files will simply cease to exist at some point in the future.  There is a ticket linked in the comment above with more information if anyone ever works on this again.